modules/mod-pcm/GetAcidizerTags.cpp

```diff
@@ -0,0 +1,77 @@
+/*  SPDX-License-Identifier: GPL-2.0-or-later */
+/*!********************************************************************
+
+  Audacity: A Digital Audio Editor
+
+  GetAcidizerTags.cpp
+
+  Matthieu Hodgkinson
+
+**********************************************************************/
+#include "GetAcidizerTags.h"
+
+#include <algorithm>
+#include <memory>
+
+namespace ModPcm
+{
+std::optional<LibFileFormats::AcidizerTags> GetAcidizerTags(
+   SNDFILE& file, const std::vector<std::string>& trustedDistributors)
+{
+   if (trustedDistributors.empty())
+      return {};
+
+   SF_LOOP_INFO loopInfo;
+   if (
+      sf_command(&file, SFC_GET_LOOP_INFO, &loopInfo, sizeof(loopInfo)) ==
+      SF_FALSE)
+      return {};
+
+   // There is loop info, but we can't trust it if the file is not from a
+   // trusted distributor:
+   SF_CHUNK_INFO info;
+   constexpr auto listId = "LIST";
+   strcpy(info.id, listId);
+   info.id_size = sizeof(listId);
+   auto chunkIt = sf_get_chunk_iterator(&file, &info);
+   while (chunkIt)
+   {
+      if (sf_get_chunk_size(chunkIt, &info) != SF_ERR_NO_ERROR)
+         break;
+      const auto chars = std::make_unique<char[]>(info.datalen);
+      info.data = chars.get();
+      if (sf_get_chunk_data(chunkIt, &info) != SF_ERR_NO_ERROR)
+         break;
+      chunkIt = sf_next_chunk_iterator(chunkIt);
+
+      const auto id = std::string { chars.get(), chars.get() + 4 };
+      if (id != "INFO")
+         continue;
+      // Next four bytes are the info tag ID:
+      const auto tagId = std::string { chars.get() + 4, chars.get() + 8 };
+      if (tagId != "IDST")
+         continue;
+      if (info.datalen < 12)
+         continue;
+      const auto size = info.datalen - 12;
+      const auto distributor =
+         std::string { chars.get() + 12, chars.get() + 12 + size };
+      const auto isTrusted = std::any_of(
+         trustedDistributors.begin(), trustedDistributors.end(),
+         [&](const auto& trustedDistributor) {
+            // trustedDistributor begins with distributor:
```

Comment:
The comment is inconsistent with the code, which is right?

You might try `std::mismatch` and might then not need the length comparison -- you expect just to get the end iterator of the prefix.

Reply:
I found an easier approach: remove trailing nulls (due to byte-padding for word alignment it would seem) and exactly compare the strings.